File: snippets/fsharp/System/Type/IsAssignableFrom/IsAssignableFrom2.fs
// <Snippet2>
open System.IO

type GenericWithConstraint<'T when 'T :> Stream>() = class end

let t = typeof<Stream>
let genericT = typeof<GenericWithConstraint<_>>.GetGenericTypeDefinition()
let genericParam = genericT.GetGenericArguments()[0]
printfn $"{t.IsAssignableFrom genericParam}"
// Displays True.
// </Snippet2>

File: snippets/csharp/System/Type/IsAssignableFrom/IsAssignableFrom2.cs
// <Snippet2>
using System;
using System.IO;

public class IsAssignableFromExample2
{
    public static void Run()
    {
        Type t = typeof(Stream);
        Type genericT = typeof(GenericWithConstraint<>);
        Type genericParam = genericT.GetGenericArguments()[0];
        Console.WriteLine(t.IsAssignableFrom(genericParam));
        // Displays True.
    }
}

public class GenericWithConstraint<T> where T : Stream
{ }
// </Snippet2>
